### ISSUE TITLE
Audit `mfs_test.go`

### ISSUE THREAD

**Issue by schomatis** [ipfs/go-mfs#9]
With 21 lines of comment to 950 LoC I have literally no idea what's being tested here (and most importantly, what's *not*). We need to start dividing this file and document which aspects of MFS it tests.

Adding the `help wanted` tag for anyone who wants to start reading it and commenting here what it finds and what can be added/improved, no need to read/fix the *entire* file.

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
